src/App.spec.ts

```diff
@@ -308,6 +308,159 @@ describe('App', () => {
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action for an external_select block (app.options)
+              body: {
+                type: 'block_suggestion',
+                action_id: 'external_select_action_id',
+                channel: {},
+                user: {},
+                team: {},
+                actions: [],
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action for "data_source": "external" in dialogs (app.options)
+              body: {
+                type: 'dialog_suggestion',
+                callback_id: 'dialog_suggestion_callback_id',
+                name: 'the name',
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.ViewSubmitAction (app.view)
+              body: {
+                type: 'view_submission',
+                channel: {},
+                user: {},
+                team: {},
+                view: {
+                  callback_id: 'view_callback_id',
+                }
+              },
+              respond: noop,
+              ack: noop,
+            },
+            // TODO: https://github.com/slackapi/bolt/issues/263
+            // {
+            //   body: {
+            //     type: 'view_closed',
+            //     channel: {},
+            //     user: {},
+            //     team: {},
+            //     view: {
+            //       callback_id: 'view_callback_id',
+            //     }
+            //   },
+            //   respond: noop,
+            //   ack: noop,
+            // },
+          ];
+        }
+
+        it('should acknowledge any of possible events', async () => {
+          // Arrange
+          const ackFn = sinon.fake.resolves({});
+          const actionFn = sinon.fake.resolves({});
+          const viewFn = sinon.fake.resolves({});
+          const optionsFn = sinon.fake.resolves({});
+          const overrides = buildOverrides(withNoopWebClient());
+          const App = await importApp(overrides); // tslint:disable-line:variable-name
+          const dummyReceiverEvents = createReceiverEvents();
+
+          // Act
+          const app = new App({ receiver: fakeReceiver, authorize: sinon.fake.resolves(dummyAuthorizationResult) });
+          app.use((_args) => { ackFn(); });
+          app.action('block_action_id', ({ }) => { actionFn(); })
+          app.action({ callback_id: 'message_action_callback_id' }, ({ }) => { actionFn(); })
+          app.action({ callback_id: 'interactive_message_callback_id' }, ({ }) => { actionFn(); })
+          app.action({ callback_id: 'dialog_submission_callback_id' }, ({ }) => { actionFn(); })
+          app.view('view_callback_id', ({ }) => { viewFn(); })
```

Comment:
Should we add one for `view_cancel`? I can do this when we merge this before my branch lands.

Reply:
@shanedewael yes, we need to have it. I haven't added the one commented out yet as we didn't decide the design at that time.